pkg/util/helper/policy.go

```diff
@@ -73,3 +77,37 @@ func IsDependentClusterOverridesPresent(c client.Client, policy *policyv1alpha1.
 
 	return true, nil
 }
+
+// GetFollowedResourceSelectorsWhenMatchServiceImport get followed derived-service and endpointSlices resource selectors
+// when policy's ResourceSelectors contains ResourceSelector, whose kind is ServiceImport.
+func GetFollowedResourceSelectorsWhenMatchServiceImport(resourceSelectors []policyv1alpha1.ResourceSelector, policyNamespace string) []policyv1alpha1.ResourceSelector {
+	var addedResourceSelectors []policyv1alpha1.ResourceSelector
+
+	for _, resourceSelector := range resourceSelectors {
+		if resourceSelector.Kind != util.ServiceImportKind {
+			continue
+		}
+
+		derivedResourceSelector := policyv1alpha1.ResourceSelector{
+			APIVersion: "v1",
+			Kind:       util.ServiceKind,
+			Namespace:  policyNamespace,
+			Name:       names.GenerateDerivedServiceName(resourceSelector.Name),
+		}
+		addedResourceSelectors = append(addedResourceSelectors, derivedResourceSelector)
+
+		endpointSliceResourceSelector := policyv1alpha1.ResourceSelector{
+			APIVersion: "discovery.k8s.io/v1beta1",
+			Kind:       util.EndpointSliceKind,
+			Namespace:  policyNamespace,
```

Comment:
The namespace is empty for ClusterPropagationPolicy, in that case the selector won't match any EndpointSlices. right?

Reply:
yeah, I'm ready to modify this.